src/plugins/data/public/ui/query_string_input/language_switcher.tsx

```diff
@@ -51,6 +55,32 @@ interface Props {
 }
 
 export function QueryLanguageSwitcher(props: Props) {
+  const opensearchDashboards = useOpenSearchDashboards<IDataPluginServices>();
+  const { application } = opensearchDashboards.services;
+  const currentApp$ = application?.currentAppId$;
+  let useNewQuerySelector;
+  application?.applications$.subscribe((applications) => {
+    applications.forEach((applicationEntry) => {
+      if (applicationEntry.id === 'observability-dashboards') {
+        useNewQuerySelector = true;
+        return;
+      }
+    });
+  });
+
+  const dataExplorerOptions = [
```

Comment:
Can we add an enum for these language options like enum ENUM_LANGUAGES { DQL = "DQL" } ? And I suggest adding an id field inside options though under context of language, label no need to wrapped by i18n.

Reply:
We will still need i18n since its purpose is to individually identify the language name. e.g. `data.query.queryBar.dqlLanguageName`